### ISSUE TITLE
[FEATURE] entity field inline javadoc

### ISSUE THREAD

**Issue by kaidohallik** [jhipster/jhipster-core#89]
* **Overview of the request**

When I enter inline javadoc for entity field then this javadoc is assigned to next entity field by jhipster:import-jdl.

For given .jh file fragment:
```
entity MyEntity {
	firstAttribute String, /** Javadoc for first attribute */
	secondAttribute String
}
```
result in MyEntity.json is:
```
    "fields": [
        {
            "fieldName": "firstAttribute",
            "fieldType": "String"
        },
        {
            "fieldName": "secondAttribute",
            "javadoc": "Javadoc for first attribute",
            "fieldType": "String"
        }
    ],
```
and in final domain class:
```
    @Column(name = "first_attribute")
    private String firstAttribute;

    /**
     * Javadoc for first attribute
     *
     */
    @ApiModelProperty(value = "Javadoc for first attribute")
    @Column(name = "second_attribute")
    private String secondAttribute;
```
Conclusion: inline javadoc for entity fields is not supported.
My feature request is: support inline javadoc for entity fields.
Note: my first feeling after encountering this was that this is a bug, but now I think it's rather feature request.

* **Motivation for or Use Case** 

Inline javadoc for entity fields saves space and keeps description compact in JDL Studio.

* **Browsers and Operating System** 

* **Related issues** 

* **Suggest a Fix** 

I'll do PR for fix soon.

**Comment by MathieuAA** [comment]
There is no issue.
```
entity MyEntity {
	firstAttribute String, /** Javadoc for first attribute */
	secondAttribute String
}
```
is wrong because of the comma. The JDL parser doesn't care about newlines and such, it only cares if there's a comma somewhere.
The correct version is:
```
entity MyEntity {
	/** Javadoc for first attribute */ firstAttribute String, 
	secondAttribute String
}
```

**Comment by kaidohallik** [comment]
It's a common practice to write inline comments at the end of the line. And it's more readable.
If I started commenting in JDL Studio then I automatically put inline comments at the end of the line.
I believe so are doing the majority of JDL Studio users.
So I suggest to add this feature.
One possible implementation of this feature is in jhipster/jhipster-core#90

**Comment by MathieuAA** [comment]
The problem is that the doc's placement is wrong. It just doesn't make any sense to add documentation for something after this something has been declared AND right before another thing is going to be declared.
That, and I don't want invisible characters to "rule the JDL world" (invisible = line break).

**Comment by kaidohallik** [comment]
I think you are doing great job and I don't want to fight for that feature (anyway I can use this feature, generating entites with code in my branch), just sending note about your last comment.

You said that you don't want line break rule the JDL world. But JDL internal comments are ruled by line break. Fragment from grammar.txt:
```
// a completely ignored comment, will not be a Javadoc comment
JDLComment = "//" [^\n\r]*
```
So there is already precedent where line break rules the JDL world.

**Comment by MathieuAA** [comment]
What if there's no comma anywhere? Except when doing things on a single line (like declaring multiple fields). That would solve the problem, don't you think?

**Comment by MathieuAA** [comment]
I'm not against it at all. The problem is the comma here. Having comments after the comma seems like commenting the next thing.

Le 1 décembre 2016 21:17:17 GMT+01:00, Kaido Hallik <notifications@github.com> a écrit :
>I think you are doing great job and I don't want to fight for that
>feature (anyway I can use this feature, generating entites with code in
>my branch), just sending note about your last comment.
>
>You said that you don't want line break rule the JDL world. But JDL
>internal comments are ruled by line break. Fragment from grammar.txt:
>```
>// a completely ignored comment, will not be a Javadoc comment
>JDLComment = "//" [^\n\r]*
>```
>So there is already precedent where line break rules the JDL world.
>
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/jhipster/jhipster-core/issues/89#issuecomment-264282405

-- 
Mathieu ABOU-AICHI

**Comment by MathieuAA** [comment]
Well, I was wrong. You can omit commas. I guess it was implemented by Carl way before the JDL was even released, and I totally missed it. My bad! 
So, to sum up, this is possible:
```
entity A {
  name String
  thing Integer
  another Double
}
```
I'm going to add the comments you requested, my bad!
This is going to be possible:
```
entity A {
  name String              /** abc */
  thing Integer /*def*/
  /*ghi*/ another Double
}
```
With this output:
```
            {
               "name": "name",
               "type": "String",
               "validations": [],
               "javadoc": " abc "
            },
            {
               "name": "thing",
               "type": "Integer",
               "validations": [],
               "javadoc": "def"
            },
            {
               "name": "another",
               "type": "Double",
               "validations": [],
               "javadoc": "ghi"
            }
```
The first comment will always override the second (`com||com2`) in the grammar.

**Comment by MathieuAA** [comment]
However, having a comma THEN a comment will result in the comment targeting the second field. You have to omit the comma to have a following comment.

**Comment by MathieuAA** [comment]
It's implemented. I'll update the docs.

**Comment by kaidohallik** [comment]
This description of the feature looks good to me. Thank you!

**Comment by kaidohallik** [comment]
I tested this and found an issue.
JDL fragment:
```
entity TestEntity {
    /** first comment */
    first String
    /** second comment */
    second String
}
```
Result in the domain class:
```
    /**
     * first comment
     */
    @ApiModelProperty(value = "first comment")
    @Column(name = "first")
    private String first;

    @Column(name = "second")
    private String second;
```
Seems that "second comment" is parsed as com2 of the field "first" and dropped.

**Comment by MathieuAA** [comment]
@kaidohallik Fixed in #97
